app/src/main/java/org/wikipedia/analytics/eventplatform/EventPlatformClient.java

```diff
@@ -0,0 +1,451 @@
+     * QUEUE may grow beyond WAIT_ITEMS.
+     *
+     * Inputs: network connection state on/off, connection state bad y/n?
+     * Taken out of iOS client, but flag can be set on the request object to wait until connected to send
+     */
+    private static boolean ENABLED = isOnline();
+
+
+    // A regular expression to match JavaScript regular expression literals. (How meta!)
+    // This is not as strict as it could be in that it allows individual flags to be specified more
+    // than once, but it doesn't really matter because we don't expect flags and ignore them if
+    // present.
+    static String JS_REGEXP_PATTERN = "^/.*/[gimsuy]{0,6}$";
+
+    /**
+     * Get the stream config for a given stream name.
+     *
+     * Stream configuration keys can take the form of either a string literal or a (JavaScript)
+     * regular expression pattern. To take advantage of the performance strengths of HashMaps, we'll
+     * first attempt to retrieve the stream config as a literal key name. If no match is found,
+     * we'll then iterate over the keys to search for a regular expression matching the provided
+     * stream name.
+     *
+     * Regex-formatted keys use JavaScript regular expression literal syntax, e.g.: '/foo/'.
+     * We don't expect any flags, and ignore them if present.
+     *
+     * N.B. Since stream config keys can be given as regular expressions, it is technically
+     * possible that more than one key could match the provided stream name. In this event that more
+     * than one match is present, we'll return the config corresponding to the first match found.
+     *
+     * @param streamName stream name
+     * @return the first matching stream config, or null if no match is found
+     */
+    static StreamConfig getStreamConfig(String streamName) {
+        if (STREAM_CONFIGS.containsKey(streamName)) {
+            return STREAM_CONFIGS.get(streamName);
+        }
+
+        for (String key : STREAM_CONFIGS.keySet()) {
+            if (key.matches(JS_REGEXP_PATTERN)) {
+                // Note: After splitting on the slash character ("/"), element [0] of the resulting
+                // array will contain the empty string (""), and element [1] will contain the
+                // regular expression pattern.
+                // If any flags are specified, they will be present in element [2], but will be
+                // ignored here.
+                if (streamName.matches(key.split("/")[1])) {
+                    return STREAM_CONFIGS.get(key);
+                }
+            }
+        }
+
+        return null;
+    }
+
+    static void setStreamConfig(StreamConfig streamConfig) {
+        STREAM_CONFIGS.put(streamConfig.getStreamName(), streamConfig);
+    }
+
+    /**
+     * Set whether the client is enabled. This can react to device online/offline state as well
+     * as other considerations.
+     */
+    public static synchronized void setEnabled(boolean enabled) {
```

Comment:
This should probably be part of what is currently called 'OutputBuffer' (see notes there). Could this be called 'enable?' Should there be a 'disable?'.

Reply:
@linehan That makes sense... but its a general convention throughout the app, setting any boolean with the positive toned variable name.